galaxyproject/galaxy
# Fix invocation report to target correct workflow version.

There was a past jmchilton todo in the code base about it. Directives that might target particular versions of a workflow (so ``workflow_display`` and ``workflow_image``) now can take a ``workflow_checkpoint`` attribute and when these things are auto-populated for workflow invocations - they are populated with the right version for the invocation. Have the report show newer versions of the workflows relative to what the invocation was generated with was a known and yucky bug.

Additionally, the ``workflow_image`` and ``workflow_license`` directives added in #16672 only worked in pages and not in invocation reports. This has been fixed in this PR as well.

## How to test the changes?
(Select all options that apply)
  1. Create a workflow and make some changes and save repeatedly to get multiple versions.
  2. Create a page and embed the workflow image and display. View the page and ensure the newest workflow is display.
  3. Generate PDF and verify it is right in there.
  4. Add ``, workflow_checkpoint=1`` to the directive and make sure older versions of the workflow are now displayed when the page is viewed.

## License
